src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/MavenPackageAndDescriptionDiagnosticRule.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.apiview.processor.diagnostics.rules;
+
+import com.azure.tools.apiview.processor.analysers.util.MiscUtils;
+import com.azure.tools.apiview.processor.diagnostics.DiagnosticRule;
+import com.azure.tools.apiview.processor.model.APIListing;
+import com.azure.tools.apiview.processor.model.Diagnostic;
+import com.azure.tools.apiview.processor.model.DiagnosticKind;
+import com.azure.tools.apiview.processor.model.maven.Pom;
+import com.github.javaparser.ast.CompilationUnit;
+
+import java.util.regex.Pattern;
+
+/**
+ * Diagnostic rule that validates the Maven package name and description match the convention specified in the
+ * guidelines.
+ */
+public final class MavenPackageAndDescriptionDiagnosticRule implements DiagnosticRule {
+    private static final Pattern MAVEN_NAME = Pattern.compile("Microsoft Azure client library for .*");
+    private static final Pattern MAVEN_DESCRIPTION =
+        Pattern.compile("This package contains the Microsoft Azure .* client library");
```

Comment:
Since we are using the same diagnostic rules for Spring, ARM and Android, this rule may not apply to them.

Reply:
Yep - you should enforce common description styles for each separately.